subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AndroidPluginsSmokeTest.groovy

```diff
@@ -84,45 +82,21 @@ class AndroidPluginsSmokeTest extends AbstractPluginValidatingSmokeTest implemen
         ))
 
         when: 'first build'
-        def result = runner.build()
+        def result = runner
+            .expectDeprecationWarningIf(agpVersion.startsWith('4.0.2') || agpVersion.startsWith('4.1'),
+                "The WorkerExecutor.submit() method has been deprecated. " +
+                    "This is scheduled to be removed in Gradle 8.0. Please use the noIsolation(), classLoaderIsolation() or processIsolation() method instead. " +
+                    "See https://docs.gradle.org/${GradleVersion.current().version}/userguide/upgrading_version_5.html#method_workerexecutor_submit_is_deprecated for more details.",
+                "TODO Add followup issue")
```

Comment:
`expectLegacyDeprecationWarning` here? This deprecation is fixed in newer versions of the plugin.

Reply:
Good point.